cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorService.java

```diff
@@ -169,4 +181,32 @@ protected long runTask() {
       return terminateTime != null && terminateTime <= currentTimeInSecs;
     };
   }
+
+  private void emitForceTerminatedRunsMetrics(ProgramRunId programRunId, RunRecordDetail recordedRunRecord,
+                                              String metricName) {
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(programRunId.getNamespaceId(),
+                                                                       recordedRunRecord.getSystemArgs());
+    Map<String, String> additionalTags = getAdditionalTagsForProgramMetrics(recordedRunRecord);
+    profile.ifPresent(profileId -> emitProfileMetrics(programRunId, profileId, metricName, additionalTags));
+  }
+
+  private Map<String, String> getAdditionalTagsForProgramMetrics(RunRecordDetail runRecord) {
```

Comment:
this method isn't needed and makes the logic harder to follow. Just add the tag in emitProfileMetrics with the other tags